### ISSUE TITLE
Rename ossec.conf to manager.conf and agent.conf

### ISSUE THREAD

**Issue by TomasTurina** [wazuh/wazuh#7787]
## Description

The goal of this issue is to rename legacy `ossec.conf` configuration file in Unix/Windows Wazuh installations.

The name of the new configuration file will be different depending on whether it is a manager or agent installation:

- **Manager**: New file name will be `manager.conf`.
- **Agent**: New file name will be `agent.conf`.

Also, to avoid confusion, `agent.conf` shared configuration file should be renamed to `shared.conf`.

### Requirements:

- This change attempts to be a breaking change, so custom settings will be lost in Wazuh upgrades.
- To reduce the effects of lost settings, `ossec.conf` will be backed up in upgrades for informational purposes only.
- New configuration file name will be different for agent and manager.
- Specific agent configuration blocks will be written only in `agent.conf` file. Example: `<syscheck>`.
- Specific manager configuration blocks will be written only in `manager.conf` file. Example: `<vulnerability-detector>`.
- For local installations, `manager.conf` file will be used.
- Rename `agent.conf` shared configuration file to `shared.conf`.
- Rename `agent-template.conf` shared configuration file to `shared-template.conf`.
- Rename `gen_ossec.sh` script file to `gen_wazuh.sh`.
- All repositories that reference `ossec.conf` file and `gen_ossec.sh` script should modify these references.

### Configuration blocks distribution

- Agent configuration file: `agent.conf`
  - `<active-response>` (client side)
  - `<agent-upgrade>` (client side)
  - `<client>`
  - `<client_buffer>`
  - `<fluent-forward>`
  - `<gcp-pubsub>`
  - `<labels>` (same as manager)
  - `<localfile>`
  - `<logging>` (same as manager)
  - `<rootcheck>`
  - `<sca>`
  - `<socket>`
  - `<syscheck>`
  - `<wodle>`
    - `<wodle name="agent-key-polling">`
    - `<wodle name="aws-s3">`
    - `<wodle name="azure-logs">`
    - `<wodle name="cis-cat">`
    - `<wodle name="command">`
    - `<wodle name="docker-listener">`
    - `<wodle name="open-scap">`
    - `<wodle name="osquery">`
    - `<wodle name="syscollector">`

- Manager configuration file: `manager.conf`
  - `<active-response>` (manager side)
  - `<agentless>`
  - `<agent-upgrade>` (manager side)
  - `<alerts>`
  - `<auth>`
  - `<cluster>`
  - `<command>`
  - `<database_output>`
  - `<email_alerts>`
  - `<global>`
  - `<integration>`
  - `<labels>` (same as agent)
  - `<logging>` (same as agent)
  - `<remote>`
  - `<reports>`
  - `<rule_test>`
  - `<ruleset>`
  - `<syslog_output>`
  - `<task-manager>`
  - `<vulnerability-detector>`

**Comment by mcarmona99** [comment]
## Issue status

In the API integration tests passed by Jenkins in some checks of the related PR: https://github.com/wazuh/wazuh/pull/8072, we have discovered a random error.

### Context:

Our **API integration environment** is a Wazuh docker environment with the following architecture:

- **Cluster** with **1 master** node and **2 worker** nodes.
- **12 Agents** (ids 001 to 012) connected to a nginx load balancer: 2 of them (011, 012) have `never connected `status and other 2 of them (009, 010) have `disconnected` status.
- **Wazuh groups:**

    **default**: 001, 002, 003, 004, 005, 006, 007, 008, 009, 010
    **group1**: 001, 005, 007, 008, 009
    **group2**: 002, 005, 006, 008, 010
    **group3**: 003, 006, 007, 008

### Error description

The error discovered is a `StopIteration` error that does not give a lot of information. 

This error can be seen in this issue https://github.com/wazuh/wazuh-jenkins/issues/2500 . This issue was created to fix this error, as we thought it was a timeout problem and the solution proposed was to increase the timeout used in the API integration tests run in Jenkins.

We have reopened that issue to print more information like the docker container statuses. This will help us understand where the error came from.

After investigating it, we have discovered the problem is related to the docker healthchecks.
Sometimes, the container where the agent with ID 004 was installed was always unhealthy.

The agent's 004 ossec.log looks like this:

```
...
2021/04/30 07:15:32 wazuh-agentd: INFO: Trying to connect to server (nginx-lb/172.19.0.2:1514/tcp).
2021/04/30 07:15:32 wazuh-agentd: INFO: (4102): Connected to the server (nginx-lb/172.19.0.2:1514/tcp).
2021/04/30 07:15:32 wazuh-agentd: ERROR: (1207): Syscheck remote configuration in 'etc/shared/shared.conf' is corrupted.
```


**Why is only agent 004 unhealthy and with this error?** This agent is the only one that only appears in group `default` and nowhere else.

After entering into the master machine, we have seen that the `default` group doesn't have its `shared.conf` file. In other words, `shared.conf` is not in `/var/ossec/etc/shared/default/`.

This error, as I said before, only appears sometimes, so it appears to be a race condition regarding the `shared.conf` file inclusion.


We had seen this error after rebasing `dev-separate-conf `more times. This PR: #8390 was supposed to solve the issue, as it fixed a problem found after merging, but it didn't.

In conclusion, tests are now passing, but this race condition is present in **dev-separate-conf**. We should take this into account and investigate more if the error is repeated. The verboses included for this error that will be added in the wazuh-jenkins issue 2500 will help too.
